src/main/java/uk/gov/cslearning/catalogue/api/MediaController.java

```diff
@@ -0,0 +1,33 @@
+package uk.gov.cslearning.catalogue.api;
+
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.multipart.MultipartFile;
+import org.springframework.web.util.UriComponentsBuilder;
+import uk.gov.cslearning.catalogue.domain.media.Media;
+import uk.gov.cslearning.catalogue.service.FileUploadFactory;
+import uk.gov.cslearning.catalogue.service.MediaManagementService;
+
+@Controller
+@RequestMapping("/service/media")
+public class MediaController {
+
+    private final MediaManagementService mediaManagementService;
+    private final FileUploadFactory fileUploadFactory;
+
+    public MediaController(MediaManagementService mediaManagementService, FileUploadFactory fileUploadFactory) {
+        this.mediaManagementService = mediaManagementService;
+        this.fileUploadFactory = fileUploadFactory;
+    }
+
+    @PostMapping
+    public ResponseEntity<Void> upload(MultipartFile file, @RequestParam String container, UriComponentsBuilder builder) {
```

Comment:
INFO - Where is the custom file name? Ie `How to cook fish` rather than `howtocookfish.pdf`

Reply:
Sorry I missed the reference to custom file name. Is it a form field? Where is it documented?